pkg/controllers/secrets.go

```diff
@@ -196,3 +196,20 @@ func RenderSecretsTemplate(templateBody string, secretsMap map[string]string) st
 
 	return buffer.String()
 }
+
+func SelectSecrets(secrets map[string]string, secretsToSelect []string) (map[string]string, error) {
+	selectedSecrets := utils.FilterMap(secrets, secretsToSelect)
+	nonExistentSecretNames := []string{}
+
+	for _, secretName := range secretsToSelect {
+		if _, found := selectedSecrets[secretName]; !found {
+			nonExistentSecretNames = append(nonExistentSecretNames, secretName)
+		}
+	}
+
+	if len(nonExistentSecretNames) > 0 {
+		return selectedSecrets, fmt.Errorf("the following secrets you are trying to include do not exist in your config:\n- %v", strings.Join(nonExistentSecretNames, "\n- "))
+	}
+
+	return selectedSecrets, nil
```

Comment:
optional but calling it out as you generally want as few return statements as possible to keep the control flow easy to understand

```suggestion
	var err error
	if len(nonExistentSecretNames) > 0 {
		err := fmt.Errorf("the following secrets you are trying to include do not exist in your config:\n- %v", strings.Join(nonExistentSecretNames, "\n- "))
	}

	return selectedSecrets, err
```